server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java

```diff
@@ -121,12 +126,23 @@ Keywords fromOldKeyword(OldKeyword oldKeyword) {
         public Keywords fromFlags(Flags flags) {
             return fromSet(Stream.concat(
                     Stream.of(flags.getUserFlags())
-                        .map(Keyword::new),
+                        .map(this::asKeyword)
+                        .filter(Optional::isPresent)
+                        .map(Optional::get),
                     Stream.of(flags.getSystemFlags())
                         .map(Keyword::fromFlag))
                 .collect(Guavate.toImmutableSet()));
         }
 
+        private Optional<Keyword> asKeyword(String flagName) {
+            try {
+                return Optional.of(new Keyword(flagName));
+            } catch (Exception e) {
+                LOGGER.warn("Fail to parse {} flag", flagName);
+                return Optional.empty();
```

Comment:
I get bored at reading this kind of code. It's time for a Try object !
what about https://github.com/poetix/ambivalence ?

Reply:
This one seems great too:
https://github.com/lambdista/try

We have to vote on this point, then I will update the PR.